obspy/scripts/runtests.py

```diff
@@ -3,774 +3,176 @@
-        # don't ask to send a report
-        args.dontask = True
-        del conf['loggers']
-    elif verbosity == 1:
-        # show all NumPy warnings
-        np.seterr(all='print')
-        # ignore user warnings
-        warnings.simplefilter("ignore", UserWarning)
-        del conf['loggers']
-    elif verbosity == 2:
-        verbosity = 1
-        # raise all NumPy warnings
-        np.seterr(all='warn')
-    else:
-        verbosity = 2
-        np.seterr(all='warn')
-        conf['loggers']['obspy']['level'] = 'DEBUG'
-    logging.config.dictConfig(conf)
-    # whether to raise any warning that's appearing
-    if args.raise_all_warnings:
-        # raise all NumPy warnings
-        np.seterr(all='raise')
-        # raise user warnings
-        warnings.simplefilter("error", UserWarning)
-        if verbosity == 2:
-            warnings.simplefilter("error", DeprecationWarning)
-    # ignore specific warnings
-    msg = ('Matplotlib is currently using agg, which is a non-GUI backend, '
-           'so cannot show the figure.')
-    warnings.filterwarnings("ignore", message=msg)
-    # check for send report option or environmental settings
-    if args.report or 'OBSPY_REPORT' in os.environ.keys():
-        report = True
-    else:
-        report = False
-    if 'OBSPY_REPORT_SERVER' in os.environ.keys():
-        args.server = os.environ['OBSPY_REPORT_SERVER']
-    # check interactivity settings
-    if interactive and args.dontask:
-        interactive = False
-    if args.keep_images:
-        os.environ['OBSPY_KEEP_IMAGES'] = ""
-    if args.keep_only_failed_images:
-        os.environ['OBSPY_KEEP_ONLY_FAILED_IMAGES'] = ""
-    if args.no_flake8:
-        os.environ['OBSPY_NO_FLAKE8'] = ""
-
-    # All arguments are used by the test runner and should not interfere
-    # with any other module that might also parse them, e.g. flake8.
-    sys.argv = sys.argv[:1]
-    return run_tests(verbosity, args.tests, report, args.log, args.server,
-                     args.test_all_modules, args.timeit, interactive, args.n,
-                     exclude=args.exclude, tutorial=args.tutorial,
-                     hostname=args.hostname, ci_url=args.ci_url,
-                     pr_url=args.pr_url)
+        # only include unique warnings.
+        data['warnings'] = [
+            dict(t) for t in {tuple(d.items()) for d in data['warnings']}
+        ]
+        response = requests.post(f"https://{REPORT_URL}/post/v2/", json=data)
+        # get the response
+        if response.status_code == 200:
+            report_url = response.json().get('url', REPORT_URL)
+            print('Your test results have been reported and are available at: '
+                  '{}\nThank you!'.format(report_url))
+        # handle errors
+        else:
+            print(f"Error: Could not sent a test report to {REPORT_URL}.")
+            print(response.reason)
 
 
-def main(argv=None, interactive=True):
+def run_tests(network=False,
```

Comment:
Can this function be removed?

Reply:
Ya, I guess we aren't really supporting running the tests from python anymore; just pytest or obspy-runtests from the command line.